src/khoj/routers/api_config.py

```diff
@@ -290,6 +290,38 @@ async def get_indexed_data_size(request: Request, common: CommonQueryParams):
     )
 
 
+@api_config.post("/user/name", status_code=200)
+@requires(["authenticated"])
+def set_user_name(
+    request: Request,
+    name: str,
+    client: Optional[str] = None,
+):
+    user = request.user.object
+
+    split_name = name.split(" ")
+
+    if len(split_name) > 2:
```

Comment:
What about folks with a middle name(s)? 😄 I think in that case the `first_name = ' '.join(split_name[:-1]) `and `last name = split_name[-1]` or vice versa (i.e `first_name = split_name[0]` and `last_name = ' '.join(split_name[1:])`)